### PR TITLE
feat: sales report package

### PR DESCRIPTION
**Pull request by Akirtovskis**

### CODE REVIEW

File: packages/sales-report-tool/index.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    await fetch(
+      `https://stacks-node-api.mainnet.stacks.co/extended/v1/address/${address}/transactions?offset=${
+        indexOfLastTxn - 20
+      }`
+    )
+      .then((response) => response.json())
+      .then((data) => {
+        data.results &&
+          data.results.length > 0 &&
+          data.results.map((txn) => {
```
Comment (pradel) [inline packages/sales-report-tool/index.js:43]: Could be simplified with
```suggestion
        data?.results?.map((txn) => {
```

File: packages/sales-report-tool/index.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+              txn.post_conditions[0]
+            ) {
+              nftsSold =
+                nftsSold + txn.post_conditions[0].amount / 1000000 / mintPrice;
+              arrOfTxn.push({
+                number: index,
+                time: getFormattedDate(txn.burn_block_time),
+                txnId: txn.tx_id,
+                address: txn.sender_address,
+                amountOfSTX: txn.post_conditions[0].amount / 1000000,
```
Comment (pradel) [inline packages/sales-report-tool/index.js:56]: relying on the postcondition might not be the best idea there, would be better to use the function name and map it to the correct number.
Could also mean that the function can be simplified and you can handle both the contract and cli calls at the same time if I get it properly

**Review by pradel — COMMENTED** [review]
@Akirtovskis code is working, just a couple of comment to simplify a bit the execution and align with the other tools we are using across Sigle.

Last but not least, could you use typescript? 🙏 
You can take the same setup as wallet-snapshot to make it easier :)

File: packages/sales-report-tool/index.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  let index = 1;
+  let nftsSold = 0;
+
+  while (indexOfLastTxn > 0 && indexOfLastTxn <= total) {
+    await fetch(
+      `https://stacks-node-api.mainnet.stacks.co/extended/v1/address/${address}/transactions?offset=${
+        indexOfLastTxn - 20
+      }`
+    )
+      .then((response: any) => response.json())
```
Comment (Akirtovskis) [inline packages/sales-report-tool/index.ts:41]: @pradel are there any ways around of getting types other than manually adding them by hand?

Comment (pradel) [reply packages/sales-report-tool/index.ts:41]: The only way to get typed reponse would be to use https://github.com/hirosystems/stacks-blockchain-api/blob/master/client/README.md

Comment (Akirtovskis) [reply packages/sales-report-tool/index.ts:41]: Should I use it or leave it w/o types here?

Comment (pradel) [reply packages/sales-report-tool/index.ts:41]: I guess it's fine

Comment (Akirtovskis) [reply packages/sales-report-tool/index.ts:41]: Good to merge then?

**Review by pradel — APPROVED** [review]

**Pull request merged**